File: src/main/java/io/github/malczuuu/iemu/configuration/ProfileSelector.java
package io.github.malczuuu.iemu.configuration;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class ProfileSelector {

  private final String[] args;

  public ProfileSelector(String[] args) {
    this.args = args;
  }

  public String getProfileName() {
    if (args.length == 1 && args[0].startsWith("--")) {
      String profile = readProfileName();
      validateProfile(profile);
      return profile;
    } else if (args.length > 1) {
      shutdown("Too many program arguments; expecting either just one [ --{profile} ] or none");
    }
    return "";
  }

  private String readProfileName() {
    return args[0].substring(2);
  }

  private void validateProfile(String profile) {
    if (profile.isEmpty()) {
      shutdown("Profile name cannot be empty");
    }
  }

  private void shutdown(String error) {
    log.error(error);
    System.exit(-1);
  }
}

File: src/main/kotlin/io/github/malczuuu/iemu/common/ProfileSelector.kt
package io.github.malczuuu.iemu.common

class ProfileSelector(private val args: Array<String>) {

  fun getProfileName(): String {
    if (args.size == 1 && args[0].startsWith("--")) {
      val profile = args[0].substring(2)
      if (profile.isEmpty()) {
        throw InvalidProfileException("Profile name cannot be empty")
      }
      return profile
    } else if (args.size > 1) {
      throw InvalidProfileException(
          "Too many program arguments; expecting either just one [ --{profile} ] or none"
      )
    }
    return ""
  }
}

class InvalidProfileException(message: String) : RuntimeException(message)
